### PR TITLE
chore(docs): add client-side support for magic links using Next.js

### PR DESCRIPTION
**Pull request by halvaradop**
## ☕️ Reasoning
This pull request adds code to the `getting-started/authentication` section for implementing magic links. It allows users to authenticate using magic links, providing a secure and user-friendly login method. The code demonstrates the client-side implementation in Next.js.

## 🧢 Checklist

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
See https://github.com/nextauthjs/next-auth/pull/11701#discussion_r1743351168

**Comment by halvaradop** [comment]
I have made changes similar to those in PR #11701. I added `Object.fromEntries` because the `signIn` function expects a value of type `SignInOptions`, which is an object. Additionally, I included the `"use client"` directive to indicate that the component works on the client side. We can discuss further if you have any questions or concerns about these changes.

File: docs/pages/getting-started/authentication/email.mdx [hunk 1]
```diff
@@ -174,6 +174,31 @@ export function SignIn() {
...
+<Code.NextClient>
+
+```tsx filename="./components/sign-in.tsx"
+"use client"
+import { signIn } from "next-auth/react"
+
+export function SignIn() {
+
+  const resendAction = (formData: formData) => {
+    signIn("resend", Object.fromEntries(formData))
```
Comment (balazsorban44) [inline docs/pages/getting-started/authentication/email.mdx:186]:
```suggestion
  const resendAction = (formData: FormData) => {
    signIn("resend", formData)
```

You can pass FormData in, we convert it automatically https://github.com/nextauthjs/next-auth/blob/965f41e0d177ed6e5e30e9b48b6ea081f2e99194/packages/next-auth/src/lib/actions.ts#L23

File: docs/pages/getting-started/authentication/email.mdx [hunk 2]
```diff
@@ -328,6 +353,31 @@ export function SignIn() {
...
+<Code.NextClient>
+
+```tsx filename="./components/sign-in.tsx"
+"use client"
+import { signIn } from "next-auth/react"
+
+export function SignIn() {
+
+  const sendgridAction = (formData: formData) => {
+    signIn("sendgrid", Object.fromEntries(formData))
```
Comment (balazsorban44) [inline docs/pages/getting-started/authentication/email.mdx:365]:
```suggestion
  const sendgridAction = (formData: FormData) => {
    signIn("sendgrid", formData)
```

**Pull request merged**